### ISSUE TITLE
Get from any start state to goal state with value iteration

### ISSUE THREAD

**Issue by cypressf** [FarmDragon/AppleCart#9]

**Comment by cypressf** [comment]
For now I added the code from the value iteration homework and the textbook example from chapter 7: https://github.com/FarmDragon/AppleCart/commit/14eee4f6220f17a5629db85faeb77811a8333d38

I'm working on branch value-iteration https://github.com/FarmDragon/AppleCart/tree/value-iteration

**Comment by cypressf** [comment]
The homework has a single-pendulum cart-pole optional exercise. I'm trying to get that to work first since I imagine it's much more feasible.

My first few runs of it while twiddling the hyperparameters didn't amount to anything. The cart just slowly moved to the right. I'm going to try visualizing the loss and the value function over time to see if I can debug what's going wrong.

**Comment by cypressf** [comment]
In our meeting with Alex, he said it's difficult to debug neural nets, but he recommends first constraining the input space to just the region around the fixed point, and testing to see if we can get the NN to stabilize, as opposed to swing up. If it can't even stabilize, maybe there's a bug in the code.

He also said we could potentially check out the pytorch cart pole system. He said that system might "cheat" in some ways that drake doesn't.

We could also try adding more data, and more layers.
